design/recovery-internals.md

```diff
@@ -90,11 +90,17 @@ CC interface is stored in `serverDBInfo`. The CC broadcasts `serverDBInfo` to al
 
 ## Phase 3: RECRUITING
 
-Once the CC locks the cstate, it will recruit the still-alive tLogs from the previous generation for the benefit of faster recovery. The CC gets the old tLogs’ interfaces from the READING_CSTATE phase and uses those interfaces to track which old tLog are still alive, the implementation of which is in `trackRejoins()`.
+Once the CC locks the cstate, it will read the transaction state store, and recruit the still-alive tLogs from the previous generation (for
+the benefit of faster recovery).
+Recruiting active tLogs is done by sending a "lock" message to each tLog. Once a tLog is locked it will not accept new commits to ingress new data, but
+old data can still be peeked by a storage server.
+The CC gets the old tLogs’ interfaces from the READING_CSTATE phase and uses those interfaces to track which old tLogs are still accessible, the implementation of which is in `trackRejoins()`. A locked tLog will reply to the CC with its `knownCommittedVersion` and `durable version`.
 
-Once the CC gets enough tLogs, it calculates the known committed version (i.e., `knownCommittedVersion` in code). `knownCommittedVersion` is the highest version that a commit proxy tells a given tLog that it had durably committed on *all* tLogs. The CC's is the maximum of all of that. `knownCommittedVersion` is  important, because it defines the lower bound of what version range of mutations need to be copied to the new generation. That is, any versions larger than the master's `knownCommittedVersion` is not guaranteed to persist on all replicas. The CC chooses a *recovery version*, which is the minimum of durable versions on all tLogs of the old generation, and recruits a new set of tLogs that copy all data between `knownCommittedVersion + 1` and `recoveryVersion` from old tLogs. This copy makes sure data within the range has enough replicas to satisfy the replication policy.
+Once the CC gets enough tLogs, it calculates the known committed version (i.e., `knownCommittedVersion`). `knownCommittedVersion` is the highest version that a commit proxy tells a given tLog that it had durably committed on *all* "local" tLogs (tLogs in remote DCs are not included).
+The CC's finds the maximum of all the tLogs. `knownCommittedVersion` is  important, because it defines the lower bound of what version range of mutations need to be copied to the new generation. That is, any versions larger than the master's `knownCommittedVersion` is not guaranteed to persist on all replicas. The CC chooses a *recovery version*, which is the minimum of durable versions on all tLogs of the old generation, and recruits a new set of tLogs that copy all data between `knownCommittedVersion + 1` and `recoveryVersion` from old tLogs. This copy makes sure data within the range has enough replicas to satisfy the replication policy.
```

Comment:
```suggestion
The CC's finds the maximum of all the tLogs. `knownCommittedVersion` is  important, because it defines the lower bound of what version range of mutations need to be copied to the new generation. That is, any versions larger than the CC's `knownCommittedVersion` is not guaranteed to persist on all replicas. The CC chooses a *recovery version*, which is the minimum of durable versions on all tLogs of the old generation, and recruits a new set of tLogs that copy all data between `knownCommittedVersion + 1` and `recoveryVersion` from old tLogs. This copy makes sure data within the range has enough replicas to satisfy the replication policy.
```